runtime/frequency/src/lib.rs

```diff
@@ -150,6 +148,31 @@ impl Contains<RuntimeCall> for BaseCallFilter {
 	}
 }
 
+impl BaseCallFilter {
+	fn is_utility_call_allowed(call: &RuntimeCall) -> bool {
+		match call {
+			RuntimeCall::Utility(pallet_utility::Call::batch { calls, .. }) |
+			RuntimeCall::Utility(pallet_utility::Call::batch_all { calls, .. }) |
+			RuntimeCall::Utility(pallet_utility::Call::force_batch { calls, .. }) =>
+				!calls.iter().any(Self::is_blocked_call),
+			_ => false,
+		}
+	}
+
+	fn is_blocked_call(call: &RuntimeCall) -> bool {
+		match call {
+            // Block following `Pays::No` from `pallet-msa` calls
+            RuntimeCall::Msa(pallet_msa::Call::revoke_delegation_by_delegator { .. }) |
+            RuntimeCall::Msa(pallet_msa::Call::revoke_delegation_by_provider { .. }) |
+            RuntimeCall::Msa(pallet_msa::Call::delete_msa_public_key { .. }) |
+            RuntimeCall::Msa(pallet_msa::Call::retire_msa { .. }) |
+            // Block following `Pays::No` from `pallet-handles` calls
+            RuntimeCall::Handles(pallet_handles::Call::retire_handle { .. }) => true,
+            _ => false,
+        }
+	}
+}
```

Comment:
This is one of the ways to do it, looking for more patterns and design decisions

Reply:
The batch will fail if anyone of these transactions match in the list of calls sent to utility